datavalidation.go

```diff
@@ -119,18 +124,26 @@ func (dv *DataValidation) SetInput(title, msg string) {
 // worksheet cells, and use the SetSqrefDropList function to set the reference
 // for their cells.
 func (dv *DataValidation) SetDropList(keys []string) error {
+	formula, err := dv.extractListFormula(keys, formula1Name)
+	if err != nil {
+		return err
+	}
+	dv.Formula1 = formula
+	dv.Type = convDataValidationType(DataValidationTypeList)
+	return nil
+}
+
+func (dv *DataValidation) extractListFormula(keys []string, formulaName string) (string, error) {
 	formula := strings.Join(keys, ",")
 	if MaxFieldLength < len(utf16.Encode([]rune(formula))) {
-		return ErrDataValidationFormulaLength
+		return "", ErrDataValidationFormulaLength
 	}
-	dv.Formula1 = fmt.Sprintf(`<formula1>"%s"</formula1>`, formulaEscaper.Replace(formula))
-	dv.Type = convDataValidationType(typeList)
-	return nil
+	return fmt.Sprintf(`<%[2]s>"%[1]s"</%[2]s>`, formulaEscaper.Replace(formula), formulaName), nil
 }
 
 // SetRange provides function to set data validation range in drop list, only
-// accepts int, float64, or string data type formula argument.
-func (dv *DataValidation) SetRange(f1, f2 interface{}, t DataValidationType, o DataValidationOperator) error {
+// accepts int, float64, string or []string data type formula argument.
```

Comment:
There already exists a function `SetDropList` that can be used for creating drop list data validation, so I think we needn't add the same support for the `SetRange` function.

Reply:
OK